### PR TITLE
Prevent `</script`, `<!--` and `<script` injection

### PR DESCRIPTION
**Pull request by fmeum**
Currently, the JSON obtained from Cloudflare Workers KV is embedded into a `script` element without any escaping. This allows for XSS and DoS via suitable injections:

* If the JSON data contains `</script>`, the `script` element will be closed and the rest of the JSON data will be executed as JavaScript in the context of the app. This allows for stored XSS in the context of the app.
* If the JSON data contains `<!--<script>`, the `script` element will not be closed by the first (and only) `</script>` end tag and will thus not be executed. This renders the app non-functional.

See [the HTML spec](https://www.w3.org/TR/html53/semantics-scripting.html#restrictions-for-contents-of-script-elements) for the list of substring that should be escaped in script elements. 

Note that the JSON data is associated with an IP address and not a stable user identifier, so this is not Self XSS. But even if you should consider it so, please take into account that this is an official example for Cloudflare Workers KV that can easily be turned into a useful app (such as a comment section on a static webpage or a simple chat app) by only changing the cache key and the HTML template. Such an app would then directly be vulnerable to XSS and a single-request DoS attack. 

The proposed change deviates from the HTML spec above to produce valid JSON. The change should be propagated to [the developer documentation](https://developers.cloudflare.com/workers/tutorials/build-a-todo-list) and the [KV blog post](https://blog.cloudflare.com/building-a-to-do-list-with-workers-and-kv/), ideally together with a comment that explains why this escaping is necessary.

### CODE REVIEW

**Review by kristianfreeman — APPROVED** [review]
lgtm, will wait for merge on the workers-docs repo before merging this corresponding one

**Comment by fmeum** [comment]
> lgtm, will wait for merge on the workers-docs repo before merging this corresponding one

Thanks!

Could you take a look into my HackerOne report at https://hackerone.com/reports/890801? It was closed as N/A, but I think that the person who triaged it did not take the context of this "demo app" into account.

**Comment by kristianfreeman** [comment]
shared this internally, there are a few things we need to rewrite here that probably necessitate a larger rewrite. i've opened a ticket in our docs https://github.com/cloudflare/workers-docs/issues/914

**Pull request merged**
